### DISCUSSION TITLE
Multiple relation field to users collection only allow to add / remove self

### DISCUSSION THREAD

**Discussion by LickABrick** [pocketbase/pocketbase#6027] [Q&A] [answered]
I have a multiple relation field to the users collection. How do I allow the user to only append / remove themself from this relation field?

I can't figure out what rule to use. I tried:
`@request.auth.id != "" && @request.body.liked_by = @request.auth.id`

**Comment by ganigeorgiev** [comment, marked as answer]
Unless I'm misunderstanding your description, there is no easy way to do what you want with the API rules.

You'll have to either use the [`OnRecordCreateRequest`](https://pocketbase.io/docs/go-event-hooks/#onrecordcreaterequest)/[`OnRecordUpdateRequest`](https://pocketbase.io/docs/go-event-hooks/#onrecordupdaterequest) hooks OR to change your collections structure to a more traditional SQL setup (aka. convert the multiple relation into a junction collection with 2 single `relation` fields).

**Reply by LickABrick** [reply]
Thanks for your response. I thought this would be the way. Now I do have another question abouw how I make sure a user can only 'like' a post once.

I currenlty have the 'likes' collection with a relation field to 'users' and 'posts'.

**Reply by ganigeorgiev** [reply]
You can specify a unique index for the pair from the "Unique constraints and indexes" section right under the collection fields.
